### DISCUSSION TITLE
Experiment not saving data to server

### DISCUSSION THREAD

**Discussion by bblankinship** [jspsych/jsPsych#746] [Q&A]
Hello!
I am relatively new to jsPsych (I have taken a short course through my department and attempted some self-learning), so it very well could be a simple mistake, but I cannot seem to get my experiment to save the data to the server (via CyberDuck). The experiment runs locally and will save data locally, but once on the server it does not save data to the server, though, there are not any error messages in Chrome or Firefox at the save data stage. 

I am at a loss! Any help or insights would be greatly appreciated!! I have included by PHP code and saveData functions and call below, please let me know if any other code would be helpful. 

```php
  $json = file_get_contents('php://input');
  $obj = json_decode($json, true);
  $outfile = fopen('/home/s1630909/server_data/'.$obj["filename"], "a");
  fwrite(
      $outfile,
      sprintf($obj["filedata"])
  );
  fclose($outfile);
```
```js
    function saveData(name, data_in){
        var url = 'record_result.php';
        var data_to_send = {filename: name, filedata: data_in};
        fetch(url, {
          method: 'POST',
          body: JSON.stringify(data_to_send),
          headers: new Headers({
              'Content-Type': 'application/json'
            })
          });
        }
```

```js
   jsPsych.init({
                timeline: timeline,
                preload_audio: audio,
                on_finish: function() {
                    var experiment_data = jsPsych.data.get();
                    saveData("results.csv", experiment_data.csv());
                }
             });
```

**Comment by jodeleeuw** [comment]
Hello! Can you please share the record_results.php code too?

**Comment by vijaymarupudi** [comment]
Could you investigate the POST request using either [Firefox] or [Chrome] developer tools and report back? Specifically, the response code and the response are of interest.

[Chrome]: https://developers.google.com/web/tools/chrome-devtools/network
[Firefox]: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/request_details

**Comment by bblankinship** [comment]
Ah sorry, I thought I had. I have just edited the post to show the PHP code

**Comment by bblankinship** [comment]
> Could you investigate the POST request using either [Firefox](https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/request_details) or [Chrome](https://developers.google.com/web/tools/chrome-devtools/network) developer tools and report back? Specifically, the response code and the response are of interest.

Thank you for the link on how to do that (I must admit I am a bit out of my depth here). I have done that I think and attached a photo here. Under "Response" it says "No response data available for this request."
<img width="1179" alt="Screenshot 2020-06-07 at 19 40 24" src="https://user-images.githubusercontent.com/53411038/83988034-e83db200-a8f6-11ea-849a-6056b9d419d2.png">
<img width="1176" alt="Screenshot 2020-06-07 at 19 41 07" src="https://user-images.githubusercontent.com/53411038/83988041-ebd13900-a8f6-11ea-9d37-de31b6e7db4c.png">

**Comment by vijaymarupudi** [comment]
Hmm, the response has no errors. I think this might be a PHP thing, that I don't have much experience with. Is there a reason you're using sprintf instead of just the filedata? Can you make your code look more like the [jsPsych example](https://www.jspsych.org/overview/data/#storing-data-permanently-as-a-file)?
